modules/apps/translation/translation-api/src/main/java/com/liferay/translation/translator/BaseTranslator.java

```diff
@@ -18,19 +19,52 @@
 @ProviderType
 public abstract class BaseTranslator implements Translator {
 
-	protected String getLanguageCode(String languageId) {
-		List<String> parts = StringUtil.split(languageId, CharPool.UNDERLINE);
+	protected String getLanguageCode(
+		boolean deepLTargetLanguage, String languageId) {
```

Comment:
Base classes shouldn't have parameters that are specific to particular implementations. Instead of this, why not override the method in the deepl impl and do there the special things for it? That will remove duplication, will keep everything in its place and won't break API:
```
@Override
protected String getLanguageCode(String languageId) {
  String languageCode = super.getLanguageCode(languageId);
  if (StringUtil.startsWith(languageCode, "en") || StringUtil.startsWith(languageCode, "pt") {
    return languageCode;
  }
  else if (StringUtil.startsWith(languageCode, "zh")) {
    ...
  }
  else {
    return ...;
  }
}
```